data/campaigns/Under_the_Burning_Suns/scenarios/07b_Talking_with_Trolls.cfg

```diff
@@ -356,26 +356,11 @@
             message= _ "Darmog has never been above ground, but Darmog understand your story. A leader must protect and care for his people. Every people deserve to find their own home. If we can help you we will."
         [/message]
 
-        [if]
-            [variable]
-                name=ally_name
-                equals=Grog
-            [/variable]
-
-            [then]
-                [message]
-                    speaker=Spiritual Advisor
-                    message= _ "We may be able to help you find a way back to the sunlit lands. In our temples we do keep records of the past. We have not walked above the earth for many many generations, not since the darkness drove us underground. But we are masters of the underground lands, and we have explored many tunnels. Recently one of our scouts found a path that leads north back to the sunlit lands, I think it may be the way you are trying to go. In reward for your achievements, we will help you. Grog has volunteered to protect you and lead you back to the sunlight lands."
-                [/message]
-            [/then]
-
-            [else]
-                [message]
-                    speaker=Spiritual Advisor
-                    message= _ "We may be able to help you find a way back to the sunlit lands. In our temples we do keep records of the past. We have not walked above the earth for many many generations, not since the darkness drove us underground. But we are masters of the underground lands, and we have explored many tunnels. Recently one of our scouts found a path that leads north back to the sunlit lands, I think it may be the way you are trying to go. In reward for your achievements, we will help you. Nog has volunteered to protect you and lead you back to the sunlight lands."
-                [/message]
-            [/else]
-        [/if]
+        {SET_ALLY_UNIT}
+        [message]
+            speaker=Spiritual Advisor
+            message= _ "We may be able to help you find a way back to the sunlit lands. In our temples we do keep records of the past. We have not walked above the earth for many many generations, not since the darkness drove us underground. But we are masters of the underground lands, and we have explored many tunnels. Recently one of our scouts found a path that leads north back to the sunlit lands, I think it may be the way you are trying to go. In reward for your achievements, we will help you. $ally_unit.name has volunteered to protect you and lead you back to the sunlight lands."
```

Comment:
With the names taken out lines like this should have a `# po: speaker is a troll`. However, that points out a new problem - the `en_US` text isn't written as troll speech.

Reply:
What do you mean by that? That not all lines spoken by trolls sound like the stereotypical 'Grog smash!' bad grammar? I noticed that too, when the author sometimes uses the same lines for both ally races, dwarves and trolls. But as I said in the original post, I haven't made (or intended to make) any dialogue changes here, just consolidate the existing lines.

Reply:
That's what I mean - a spiritual advisor should will be more learned than the average, but at the same time a troll spiritual advisor probably shouldn't sound exactly the same as a dwarf spiritual advisor.

It's an issue in the existing text, but seems something that should be fixed for 1.18.

Reply:
The dwarves don't have a spiritual adviser - allying with the dwarves has Kaleh and friends speaking with a dwarvish king in S7a.

Anyway, I'm glad you agree the shared lines (between dwarf and troll allies) are a separate issue.

Reply:
Didn't add the `po`hint here because the adviser is referring to the ally, not to himself, with the `$ally_unit.name` mention.